litmus-portal/authentication/pkg/entities/user.go

```diff
@@ -8,20 +8,26 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-//User contains the user information
+// User contains the user information
 type User struct {
 	ID            string  `bson:"_id,omitempty" json:"_id"`
 	UserName      string  `bson:"username,omitempty" json:"username"`
 	Password      string  `bson:"password,omitempty" json:"password,omitempty"`
 	Email         string  `bson:"email,omitempty" json:"email,omitempty"`
 	Name          string  `bson:"name,omitempty" json:"name,omitempty"`
 	Role          Role    `bson:"role,omitempty" json:"role"`
-	LoggedIn      bool    `bson:"logged_in,omitempty" json:"logged_in,omitempty"`
 	CreatedAt     *string `bson:"created_at,omitempty" json:"created_at,omitempty"`
 	UpdatedAt     *string `bson:"updated_at,omitempty" json:"updated_at,omitempty"`
 	DeactivatedAt *string `bson:"deactivated_at,omitempty" json:"deactivated_at,omitempty"`
 }
 
+// UserDetails is used to update user's personal details
+type UserDetails struct {
+	Email    string `bson:"email,omitempty" json:"email,omitempty"`
+	Name     string `bson:"name,omitempty" json:"name,omitempty"`
+	Password string `bson:"password,omitempty" json:"password,omitempty"`
+}
+
```

Comment:
I can add uid, but we are anyways getting it from the context.

Reply:
There only reason I didn't include that in the struct was: we will have to put an extra check to validate if input uid is same as the uid of logged in user.

Reply:
Not needed? just set the uid in the struct, don't put the json tag in the uid field the user will not send the uid it will be taken from the cookie itself.